src/core/qgsrange.h

```diff
@@ -392,12 +394,66 @@ class QgsTemporalRange
       return false;
     }
 
+    /**
+     * Extends the range in place by adding an \a other range.
+     * If \a other is empty the range is not changed.
+     * If the range is empty and \a other is not, the range is changed and set to \a other.
+     * \see isEmpty()
+     * \since QGIS 3.12
+     * \return TRUE if the range was extended
+     */
+    bool extend( const QgsTemporalRange<T> &other )
+    {
+      if ( other.isEmpty() || other.isInfinite() )
+      {
+        return false;
+      }
+      else if ( isEmpty() || isInfinite() )
```

Comment:
This also looks wrong. An infinite range should not be shrunk to match a smaller one